lib/geoengineer/template.rb

```diff
@@ -5,17 +5,15 @@ class GeoEngineer::Template
   include HasAttributes
   include HasResources
 
-  def initialize(name, project, parameters = {})
-    @project = project
+  def initialize(name, environment, parameters = {})
```

Comment:
Maybe pass both project and environment at the same time, and for just environments pass project as nil. Or maybe pass @parent attribute for resources to be created on.